src/compiler.js

```diff
@@ -208,104 +289,235 @@ class ReadCompiler {
+function containerInlining (values) {
+  // Inlining (support only 1 level)
+  const newValues = []
+  for (const i in values) {
+    const { type, anon } = values[i]
+    if (anon) {
+      if (type instanceof Array && type[0] === 'container') {
+        for (const j in type[1]) newValues.push(type[1][j])
+        console.log('Inlined an anonymous container')
+      } else if (type instanceof Array && type[0] === 'switch') {
+        const theSwitch = type[1]
+        const valueSet = new Set()
+        // search for containers and build a set of possible values
+        for (const field in theSwitch.fields) {
+          if (theSwitch.fields[field] instanceof Array && theSwitch.fields[field][0] === 'container') {
+            for (const j in theSwitch.fields[field][1]) {
+              const item = theSwitch.fields[field][1][j]
+              valueSet.add(item.name)
+            }
+          }
+        }
+        // For each value create a switch
+        for (const name of valueSet.keys()) {
+          const fields = {}
+          for (const field in theSwitch.fields) {
+            if (theSwitch.fields[field] instanceof Array && theSwitch.fields[field][0] === 'container') {
+              for (const j in theSwitch.fields[field][1]) {
+                const item = theSwitch.fields[field][1][j]
+                if (item.name === name) {
+                  fields[field] = theSwitch.fields[field][1][j].type
+                  break
+                }
+              }
+            } else {
+              fields[field] = theSwitch.fields[field]
+            }
+          }
+          newValues.push({
+            name,
+            type: ['switch', {
+              compareTo: theSwitch.compareTo,
+              compareToValue: theSwitch.compareToValue,
+              default: theSwitch.default,
+              fields
+            }]
+          })
+        }
+        console.log('Inlined an anonymous switch')
+      } else {
+        throw new Error('Cannot inline anonymous type: ' + type)
       }
+    } else {
+      newValues.push(values[i])
     }
-    return '() => {\nconst ctx = {\n' + indent(Object.keys(functions).map((type) => {
-      return type + ': ' + functions[type]
-    }).join(',\n')) + '\n}\n  return ctx\n}'
   }
+  return newValues
+}
 
-  /**
-   * Compile the given js code, providing native.{type} to the context, return the compiled types
-   * @param {*} code
-   */
-  compile (code) {
-    // Local variable to provide some context to eval()
-    const native = this.native // eslint-disable-line
-    return eval(code)() // eslint-disable-line
+function swapMappings (json) {
+  const ret = {}
+  for (let key in json) {
+    ret[json[key]] = (isNaN(key)) ? key : parseInt(key, 10)
   }
+  return ret
 }
 
 function optimize (code, cb) {
   const closureCompiler = new ClosureCompiler({
-    language_in: 'ES6',
     compilation_level: 'SIMPLE'
```

Comment:
Umm, did 'ADVANCED' mode breaks code?

Reply:
Yes, it is too aggressive with variable renaming, even when specifying the correct externs. There might be a way to make it work but a lot of configuration would be required.